filament/backend/src/vulkan/VulkanHandles.cpp

```diff
@@ -403,18 +403,18 @@ bool VulkanRenderTarget::invalidate() {
 
 VulkanVertexBuffer::VulkanVertexBuffer(VulkanContext& context, VulkanStagePool& stagePool,
         VulkanDisposer& disposer,  uint8_t bufferCount, uint8_t attributeCount,
-        uint32_t elementCount, AttributeArray const& attributes) :
-        HwVertexBuffer(bufferCount, attributeCount, elementCount, attributes) {
+        uint32_t elementCount, AttributeArray const& attributes, bool boEnabled) :
+        HwVertexBuffer(bufferCount, attributeCount, elementCount, attributes, boEnabled) {
     buffers.reserve(bufferCount);
     for (uint8_t bufferIndex = 0; bufferIndex < bufferCount; ++bufferIndex) {
         uint32_t size = 0;
         for (auto const& item : attributes) {
             if (item.buffer == bufferIndex) {
-              uint32_t end = item.offset + elementCount * item.stride;
+                uint32_t end = item.offset + elementCount * item.stride;
                 size = std::max(size, end);
             }
         }
-        buffers.emplace_back(new VulkanBuffer(context, stagePool, disposer, this,
+        buffers.push_back(std::make_shared<VulkanBuffer>(context, stagePool, disposer, this,
```

Comment:
One thing I don't like about `shared_ptr` is that it's unclear where and when the destructor is called, which can be a problem if the destructor has side-effects or in a multithreaded environment.

I don't know if there would be a problem here (I'm not familiar with the vulkan code enough), but I want to call it out, so we check that all the places where a reference can drop are safe for the dtor to be called (e.g. locks are not held --or-- always held, etc...).

Reply:
Yes I was hesitant about this but I felt it was the right thing to use here, since the ownership truly is shared between two objects.  I thought about it and the destructor will be invoked at the correct time.